src/code_f8f60_len_1560.c

```diff
@@ -31,21 +31,226 @@ s32 UpdateLerp(script_context* script, s32 initialCall) {
 }
 
-INCLUDE_ASM(code_f8f60_len_1560, RandInt);
+s32 RandInt(script_context* script, s32 initialCall) {
+    bytecode* ptrReadPos = script->ptrReadPos;
 
-INCLUDE_ASM(code_f8f60_len_1560, GetAngleBetweenNPCs);
+    s32 max = get_variable(script, *ptrReadPos++);
+    bytecode outVar = *ptrReadPos++;
 
-INCLUDE_ASM(code_f8f60_len_1560, GetAngleToNPC);
+    set_variable(script, outVar, rand_int(max));
 
-INCLUDE_ASM(code_f8f60_len_1560, GetAngleToPlayer);
+    return 2;
+}
+
+s32 GetAngleBetweenNPCs(script_context* script, s32 initialCall) {
+    bytecode* ptrReadPos = script->ptrReadPos;
+
+    NPC aID = get_variable(script, *ptrReadPos++);
+    NPC bID = get_variable(script, *ptrReadPos++);
+    bytecode outVar = *ptrReadPos++;
+
+    npc* a = resolve_npc(script, aID);
+    npc* b = resolve_npc(script, bID);
+    set_variable(script, outVar, atan2(a->pos.x, a->pos.z, b->pos.x, b->pos.z));
+
+    return 2;
+}
+
+s32 GetAngleToNPC(script_context* script, s32 initialCall) {
+    player_status* playerStatus = &gPlayerStatus;
+    bytecode* ptrReadPos = script->ptrReadPos;
+
+    NPC npcID = get_variable(script, *ptrReadPos++);
+    bytecode outVar = *ptrReadPos++;
+
+    npc* npc = resolve_npc(script, npcID);
+    set_variable(script, outVar, atan2(playerStatus->position.x, playerStatus->position.z, npc->pos.x, npc->pos.z));
+
+    return 2;
+}
+
+s32 GetAngleToPlayer(script_context* script, s32 initialCall) {
+    player_status* playerStatus = &gPlayerStatus;
+    bytecode* ptrReadPos = script->ptrReadPos;
+
+    NPC npcID = get_variable(script, *ptrReadPos++);
+    bytecode outVar = *ptrReadPos++;
 
-INCLUDE_ASM(code_f8f60_len_1560, AwaitPlayerApproach);
+    npc* npc = resolve_npc(script, npcID);
+    set_variable(script, outVar, atan2(npc->pos.x, npc->pos.z, playerStatus->position.x, playerStatus->position.z));
 
-INCLUDE_ASM(code_f8f60_len_1560, IsPlayerWithin);
+    return 2;
+}
+
+s32 AwaitPlayerApproach(script_context* script, s32 initialCall) {
+    bytecode* ptrReadPos = script->ptrReadPos;
+    player_status* playerStatus = &gPlayerStatus;
+
+    s32* targetX = &script->functionTemp[0];
+    s32* targetZ = &script->functionTemp[1];
+    s32* distanceRequired = &script->functionTemp[2];
+
+    f32 distance;
+
+    if (initialCall) {
+        *targetX = get_variable(script, *ptrReadPos++);
+        *targetZ = get_variable(script, *ptrReadPos++);
+        *distanceRequired = get_variable(script, *ptrReadPos++);
+    }
+
+    distance = dist2D(
+        playerStatus->position.x, playerStatus->position.z,
+        *targetX, *targetZ
+    );
+
+    if (distance < *distanceRequired) {
+        return 2;
+    } else {
+        return 0;
+    }
+}
+
+s32 IsPlayerWithin(script_context* script, s32 initialCall) {
+    bytecode* ptrReadPos = script->ptrReadPos;
+    player_status* playerStatus = &gPlayerStatus;
 
-INCLUDE_ASM(code_f8f60_len_1560, AwaitPlayerLeave);
+    /* Function temporaries are redundant, stack should be used instead */
```

Comment:
Not sure all of these redundant comments are really needed

Reply:
Yeah, I was iffy on that as well..